### PR TITLE
fix(ingest): Use lower-case dataset names in the dataset urns for all SQL-styled datasets.

### PR DESCRIPTION
**Pull request by rslanka**
Converts the dataset name to lower-case in the dataset urns for all SQL-styled datasets.

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-ingestion/src/datahub/emitter/mce_builder.py [hunk 1]
```diff
@@ -35,7 +35,20 @@
...
+    "hive",
+    "mariadb",
+    "mssql",
+    "mysql",
+    "oracle",
+    "postgres",
+    "redshift",
+    "snowflake",
+    "trino",
+}
```
Comment (gabe-lyons) [inline metadata-ingestion/src/datahub/emitter/mce_builder.py:51]: is there any more generic way to specify sql systems? What's the plan for making sure this set stays updated?

Comment (rslanka) [reply metadata-ingestion/src/datahub/emitter/mce_builder.py:51]: So long as we use the mce_builder to mint urns, there is no fool-proof way to ensure this, except via code-reviews. The other way would be to provide an urn minting factory that is customizable per source. That's a much bigger change, and we should do it at some point in future.

File: metadata-ingestion/src/datahub/emitter/mce_builder.py [hunk 2]
```diff
@@ -70,7 +86,13 @@ def make_dataplatform_instance_urn(platform: str, instance: str) -> str:
 def make_dataset_urn_with_platform_instance(
     platform: str, name: str, platform_instance: Optional[str], env: str = DEFAULT_ENV
 ) -> str:
+
     if platform_instance:
+        # Use lower-case name for all SQL style datasets
+        if platform in SQL_STYLE_PLATFORMS:
```
Comment (swaroopjagadish) [inline metadata-ingestion/src/datahub/emitter/mce_builder.py:92]: There is no need to use lower-case for platform instance

**Review by jjoyce0510 — COMMENTED** [review]
Do we expect that folks have previously ingested non-case sensitive datasets? This change will mint new datasets altogether, without any migration path. Do we have reason to believe this won't be disruptive?

**Review by shirshanka — APPROVED** [review]

**Pull request merged**